arducate/src/components/Toolbar.js

```diff
@@ -29,11 +32,82 @@ const Toolbar = () => {
     window.open(url, "_blank");
   };
 
-  const handlePublish = () => {
-    const htmlContent = convertSceneToAR(arObjects);
-    const blob = new Blob([htmlContent], { type: "text/html" });
-    const url = URL.createObjectURL(blob);
-    window.open(url, "_blank");
+  // Handle the publishing action
+  const handlePublish = async () => {
+    if (!generatedUrl) { // Only generate a new URL if one doesn't exist
+      const htmlContent = convertSceneToAR(arObjects);
+      const uniqueId = uuidv4(); // Generate a unique ID for the content
+
+      try {
+        const { data, error } = await supabase
+          .from('ar_content')
+          .insert([
+            { url: uniqueId, html_content: htmlContent }
+          ]);
+  
+        if (error) {
+          throw new Error('Failed to save content: ' + error.message);
+        }
+
+        const baseUrl = window.location.origin; 
+        const arContentUrl = `${baseUrl}/ar-content/${uniqueId}`;        setGeneratedUrl(arContentUrl); // Store the URL
```

Comment:
idk why it shows this in the same line, dunno if thats just a formatting thing, but worth checking!